fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/InferPredicates.java

```diff
@@ -156,18 +159,22 @@ private Set<Expression> getAllExpressions(Plan left, Plan right, Optional<Expres
         Set<Expression> baseExpressions = pullUpPredicates(left);
         baseExpressions.addAll(pullUpPredicates(right));
         condition.ifPresent(on -> baseExpressions.addAll(ExpressionUtils.extractConjunction(on)));
-        baseExpressions.addAll(PredicatePropagation.infer(baseExpressions));
-        return baseExpressions;
+        return PredicateInferUtils.inferPredicate(baseExpressions);
```

Comment:
seems that PredicatePropagation.infer is replaced by the new utils, can the former one be removed? Or need to replace all reference of the old one with the new util?

Reply:
the former one can be removed, all reference of the old one has been replaced.